web/src/i18n.ts

```diff
@@ -60,12 +60,29 @@ export function $t_html(
     });
 }
 
-export let language_list: (typeof page_params & {page_type: "home"})["language_list"];
+export let language_list: ((typeof page_params & {page_type: "home"})["language_list"][number] & {
+    display_name: string;
+})[];
 
 export function get_language_name(language_code: string): string | undefined {
     return language_list.find((language) => language.code === language_code)?.name;
 }
 
+function get_language_display_name(language_code: string): string {
+    const locale = new Intl.Locale(language_code === "en" ? "en-US" : language_code);
+    let language_display_name = new Intl.DisplayNames([locale], {type: "language"})
+        .of(locale.language)!
```

Comment:
Have used `!` here assuming the server would send language codes that will have a display name value. Can add a condition to show the original name if `Intl.DisplayNames` returns undefined.

Reply:
Probably fine to do it this way, failing visibly seems better here.